### ISSUE TITLE
[improvement]: Support RFC-5031 Uniform Resource Name (URN) for Emergency and Other Well-Known Services

### ISSUE THREAD

**Issue by imcdona** [asterisk/asterisk#192]
### Improvement Description

tl;dr: Add support for making/receiving calls to SIP URN's eg. `INVITE urn:service:sos`

It's important to put this request into context so you can better understand why RFC-5031 support is needed.

Prior to the advent of VoIP, emergency calls were routed to the nearest PSAP based on the physical address of the caller as provided by the phone company. In a VoIP world, or cellular for that matter, the location of the caller is dynamic. Not only is the location of the caller in question, the number used to reach emergency services will differ depending on the country from which the call is placed.

Asterisk supports the ability to include the callers location via the _geolocation_ module. This ensures the callers location is provided when placing emergency calls. The second part of this scenario, determining the number to dial to reach emergency services, is what RFC-5031 addresses.

With the addition of RFC-5031, along with the functionality provided by the _geolocation_ module, Asterisk would be capable of sending a NENA i3 SIP INVITE. This is is the missing piece required to make Asterisk compliant with next generation emergency calling.

Adding the capability to accept incoming SIP INVITES to a SIP URN would make Asterisk capable of routing i3 compliant SIP INVITES making it possible to use Asterisk as a PBX in a PSAP environment. I'm currently getting around this limitation by sending outbound calls via Kamailio which then modifies the Request-URI` $ru = "urn:service:sos";` as _PJSIP_HEADER()_ isn't capable of modifying the Request-URI.

Thanks

**Comment by imcdona** [comment]
An improvement request. Or maybe it could be considered a feature request? I don't have any code.
